packages/core/src/amazonqTest/error.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { ToolkitError } from '../shared/errors'
+
+export const technicalErrorCustomerFacingMessage =
+    'I am experiencing technical difficulties at the moment. Please try again in a few minutes.'
+const defaultTestGenErrorMessage = 'Amazon Q encountered an error while generating tests. Try again later.'
+export class TestGenError extends ToolkitError {
+    constructor(
+        error: string,
+        code: string,
+        public statusCode: string,
+        public uiMessage: string
+    ) {
+        super(error, { code })
+    }
+}
+export class ProjectZipError extends TestGenError {
+    constructor(error: string) {
+        super(error, 'ProjectZipError', '400', defaultTestGenErrorMessage)
+    }
+}
+export class InvalidSourceZipError extends TestGenError {
+    constructor() {
+        super('Failed to create valid source zip', 'InvalidSourceZipError', '400', defaultTestGenErrorMessage)
+    }
+}
+export class CreateUploadUrlError extends TestGenError {
+    constructor(errorMessage: string, errorCode: string) {
+        super(errorMessage, 'CreateUploadUrlError', errorCode, technicalErrorCustomerFacingMessage)
+    }
+}
+export class UploadTestArtifactToS3Error extends TestGenError {
+    constructor(error: string, statusCode?: string) {
+        super(error, 'UploadTestArtifactToS3Error', statusCode ?? '400', technicalErrorCustomerFacingMessage)
+    }
+}
+export class CreateTestJobError extends TestGenError {
+    constructor(error: string, code: string) {
+        super(error, 'CreateTestJobError', code, technicalErrorCustomerFacingMessage)
+    }
+}
+export class TestGenTimedOutError extends TestGenError {
```

Comment:
What is the plan here? Maintaining one-to-one errors with all the service-side errors is not maintainable. Can you find something more generalizable.

Reply:
`TestGenTimedOutError` is client side error if plugin did not get response for limited time we throw this error. This helps us understand the number of requests timed out from generic backend errors.

Reply:
I am not asking about this error only, I'm referring to all of the errors in this file. It is not maintainable to create special errors for every service-side error.